control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1655,6 +1658,69 @@ func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyp
 	return remaining, errors.NewAggregate(errs)
 }
 
+func (r *reconciler) ensureGuestAdmissionWebhooksAreValid(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	cpServices := &corev1.ServiceList{}
+	if err := r.cpClient.List(ctx, cpServices, client.InNamespace(r.hcpNamespace)); err != nil {
+		return fmt.Errorf("failed to list control plane services")
+	}
+
+	// disallow all urls tageting services in the hcp namespace by default unless 'hypershift.openshift.io/allow-guest-webhooks' label is present.
+	disallowedUrls := make([]string, 0)
+	for _, svc := range cpServices.Items {
+		if _, exist := svc.Labels["hypershift.openshift.io/allow-guest-webhooks"]; !exist {
+			disallowedUrls = append(disallowedUrls, fmt.Sprintf("https://%s:", svc.Name))
+			disallowedUrls = append(disallowedUrls, fmt.Sprintf("https://%s.%s.svc", svc.Name, svc.Namespace))
+			disallowedUrls = append(disallowedUrls, fmt.Sprintf("https://%s.%s.svc.cluster.local", svc.Name, svc.Namespace))
+		}
+	}
+
+	validatingWebhookConfigurations := &admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	if err := r.client.List(ctx, validatingWebhookConfigurations); err != nil {
+		return err
+	}
+
+	errs := make([]error, 0)
+	for _, configuration := range validatingWebhookConfigurations.Items {
+		for _, webhook := range configuration.Webhooks {
+			if webhook.ClientConfig.URL != nil && !isAllowedWebhookUrl(disallowedUrls, *webhook.ClientConfig.URL) {
+				log.Info("deleting validating webhook configuration with a disallowed url", "webhook_name", configuration.Name, "disallowed_url", *webhook.ClientConfig.URL)
+				errs = append(errs, r.client.Delete(ctx, &configuration))
+				break
+			}
+		}
+	}
+
+	mutatingWebhookConfigurations := &admissionregistrationv1.MutatingWebhookConfigurationList{}
+	if err := r.client.List(ctx, mutatingWebhookConfigurations); err != nil {
```

Comment:
This reinitializes `err` so you would lose any errors captured on L1689 right?

Reply:
why would I lose any errors?  `err` is not defined anywhere before this line.

Reply:
Yeah you're right. I misread L1684 as `err` 🤦🏻